packages/core/utils/src/schema.js

```diff
@@ -36,6 +36,7 @@ export type SchemaNot = {|
 export type SchemaString = {|
   type: 'string',
   enum?: Array<string>,
+  validate?: (val: string) => ?string,
```

Comment:
(Nit) So far, I've prefixed properties that are additions to the JSON Schema standard with `__`.
For consistency, it should either be `__validate` or we remove `__` from the other additions.

Reply:
I'll add it in a follow-up

Reply:
(wasn't merged yet when I started writing that 😄 )